server/tests/unit/mavat/mavat.test.js

```diff
@@ -0,0 +1,182 @@
+const assert = require('chai').assert;
+const nock = require('nock');
+const fs = require('fs');
+const path = require('path');
+const os = require('os');
+const crypto = require('crypto');
+
+const { downloadChallengedFile } = require('../../../api/lib/mavat').testOnly;
+
+describe('Challenged file download', function() {
+	beforeEach(async function() {
+		// make sure nock is active
+		if (!nock.isActive())
+			nock.activate();
+	});
+
+	afterEach(async function() {
+		// restore unmocked networking
+		nock.restore()
+	});
+
+	it('should handle errors properly', async function() {
+		// mock mavat unchallenged pdf file
+		const mavatScope = nock('http://mavat.moin.gov.il')
+			.get('/MavatPS/Forms/Attachment.aspx')
+			.query({
+				edid: 1,
+				edn: 'FAKEEDN',
+				opener: 'AttachmentError.aspx'
+			})
+			// reply with a not-found error
+			.reply(404, 'page not found');
+
+		// create a temporary file for the result to be written to
+		const resFile = fs.createWriteStream(path.join(os.tmpdir(), `${crypto.randomBytes(8).toString('hex')}.pdf`));
+
+		// try to download the file
+		const resSuccess = await downloadChallengedFile(
+			'http://mavat.moin.gov.il/MavatPS/Forms/Attachment.aspx?edid=1&edn=FAKEEDN&opener=AttachmentError.aspx',
+			resFile
+		);
+
+		// make sure all mocked urls were accessed
+		mavatScope.done();
+
+		// make sure the result was not successful and that nothing was written to the file
+		assert.equal(resSuccess, false, 'download should not be successful');
+		assert.equal(fs.readFileSync(resFile.path), '', 'file contents should be empty');
+
+		// delete the temporary file
+		fs.unlinkSync(resFile.path);
+	});
+
+	it('should handle non-challenged html pages', async function() {
+		// mock mavat unchallenged pdf file
+		const mavatScope = nock('http://mavat.moin.gov.il')
+			.get('/MavatPS/Forms/Attachment.aspx')
+			.query({
+				edid: 1,
+				edn: 'FAKEEDN',
+				opener: 'AttachmentError.aspx'
+			})
+			// reply with a not-found error
+			.reply(
+				200,
+				'<html><body>no challenge here</body></html>',
+				{'Content-Type': 'text/html'}
+			);
+
+		// create a temporary file for the result to be written to
+		const resFile = fs.createWriteStream(path.join(os.tmpdir(), `${crypto.randomBytes(8).toString('hex')}.pdf`));
+
+		// try to download the (un)challenged file
+		const resSuccess = await downloadChallengedFile(
+			'http://mavat.moin.gov.il/MavatPS/Forms/Attachment.aspx?edid=1&edn=FAKEEDN&opener=AttachmentError.aspx',
```

Comment:
extract all const to configuration, address, coockies etc not readable + not indicative